kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/KafkaEntryFormatter.java

```diff
@@ -57,11 +56,18 @@ public MemoryRecords decode(List<Entry> entries, byte magic) {
             return byteBuf.slice(byteBuf.readerIndex(), byteBuf.readableBytes());
         }).collect(Collectors.toList());
 
-        CompositeByteBuf batchedByteBuf = Unpooled.compositeBuffer();
-        batchedByteBuf.addComponents(true, orderedByteBuf);
+        // batched ByteBuf should be released after sending to client
+        int totalSize = orderedByteBuf.stream().mapToInt(ByteBuf::readableBytes).sum();
+        ByteBuf batchedByteBuf = PooledByteBufAllocator.DEFAULT.directBuffer(totalSize);
```

Comment:
Should we use `PulsarByteBufAllocator.DEFAULT` instead of using Netty's allocator directly?